extra/modules/pb-richmedia-filter/src/test/java/org/prebid/server/hooks/modules/pb/richmedia/filter/v1/PbRichmediaFilterBidResponsesFilterHookTest.java

```diff
@@ -0,0 +1,269 @@
+    private AllProcessedBidResponsesPayload allProcessedBidResponsesPayload;
+
+    @Mock
+    private AuctionInvocationContext auctionInvocationContext;
+
+    @Mock
+    private BidResponsesMraidFilter mraidFilter;
+
+    private PbRichmediaFilterBidResponsesFilterHook target;
+
+    @Before
+    public void setUp() {
+        target = new PbRichmediaFilterBidResponsesFilterHook(ObjectMapperProvider.mapper(), mraidFilter, true);
+    }
+
+    @Test
+    public void shouldHaveValidInitialConfigs() {
+        // given & when & then
+        assertThat(target.code()).isEqualTo("pb-richmedia-filter-bid-responses-filter-hook");
+    }
+
+    @Test
+    public void callShouldReturnResultWithNoActionWhenFilterMraidIsFalse() {
+        // given
+        target = new PbRichmediaFilterBidResponsesFilterHook(ObjectMapperProvider.mapper(), mraidFilter, false);
+        final List<BidderResponse> givenResponses = givenBidderResponses(2);
+        doReturn(givenResponses).when(allProcessedBidResponsesPayload).bidResponses();
+
+        // when
+        final Future<InvocationResult<AllProcessedBidResponsesPayload>> future = target.call(
+                allProcessedBidResponsesPayload,
+                auctionInvocationContext);
+
+        // then
+        assertThat(future).isNotNull();
+        assertThat(future.succeeded()).isTrue();
+
+        final InvocationResult<AllProcessedBidResponsesPayload> result = future.result();
+        assertThat(result).isNotNull();
+        assertThat(result.status()).isEqualTo(InvocationStatus.success);
+        assertThat(result.action()).isEqualTo(InvocationAction.no_action);
+        assertThat(result.analyticsTags()).isNull();
+        assertThat(result.payloadUpdate().apply(AllProcessedBidResponsesPayloadImpl.of(List.of())).bidResponses())
+                .isEqualTo(givenResponses);
+
+        verifyNoInteractions(mraidFilter);
+    }
+
+    //todo: is the reject possible?
+/*    @Test
+    public void callShouldReturnResultWithRejectActionWhenAllBidderResponsesWereFilteredOut() {
+        // given
+        final List<BidderResponse> givenResponses = givenBidderResponses(2);
+        doReturn(givenResponses).when(allProcessedBidResponsesPayload).bidResponses();
+        given(mraidFilter.filter(givenResponses))
+                .willReturn(MraidFilterResult.of(Collections.emptyList(), Collections.emptyList()));
+
+        // when
+        final Future<InvocationResult<AllProcessedBidResponsesPayload>> future = target.call(
+                allProcessedBidResponsesPayload,
+                auctionInvocationContext);
+
+        // then
+        assertThat(future).isNotNull();
+        assertThat(future.succeeded()).isTrue();
+
+        final InvocationResult<AllProcessedBidResponsesPayload> result = future.result();
+        assertThat(result).isNotNull();
+        assertThat(result.status()).isEqualTo(InvocationStatus.success);
+        assertThat(result.action()).isEqualTo(InvocationAction.reject);
+    }*/
```

Comment:
From what i've seen in your code this is not possible at all, so please, remove.
P.S. Also, reject will reject all bid responses in that case.

Reply:
Yeah, I know, I don't question my implementation, I question the discussion with @SerhiiNahornyi that rejection case is possible, but eventually I don't see it possible